migrations/3_deploy_node_ingress_rules_contract.js

```diff
@@ -12,10 +12,22 @@ const rulesContractName = Web3Utils.utf8ToHex("rules");
 let nodeIngress = process.env.NODE_INGRESS_CONTRACT_ADDRESS;
 let retainCurrentRulesContract = AllowlistUtils.getRetainNodeRulesContract();
 
+async function logCurrentAllowlist(instance) {
+    let s = await instance.getSize();
```

Comment:
Could this be optimised to one async call to get the whole collection, instead of an async call every iteration?

Reply:
solidity functions can't return arrays of unbounded objects (ie strings) so for nodes we need to get them one at a time